tests/contracts/Polynetwork.scilla

```diff
@@ -0,0 +1,303 @@
+scilla_version 0
+
+import Polynetwork BoolUtils
+
+library Polynetwork_local
+
+let nullAddress = 0x0000000000000000000000000000000000000000
+
+let one_msg =
+  fun (msg : Message) =>
+    let nil_msg = Nil {Message} in
+    Cons {Message} msg nil_msg
+
+contract Polynetwork(thisChainID : Uint64)
```

Comment:
As far as I can tell this contract corresponds to the Solidity contract, but I'm not 100% certain because much of the library code is complete magic to me.

I'm concerned about how we will maintain this code. As far as I can tell we need to be downstream from the Solidity contract, but I don't understand the process by which that happens.

Reply:
There doesn't seem to be an answer to this. Presumably we just have to trust that we'll be notified if something needs to be fixed. It's not ideal TBH, but I suppose it will have to do.